lib/spack/spack/operating_systems/mac_os.py

```diff
@@ -150,9 +150,8 @@ def __init__(self):
         # only used the minor component)
         part = 1 if version >= Version('11') else 2
 
-        mac_ver = str(version.up_to(part))
-        name = mac_releases.get(mac_ver, "macos")
-        super(MacOs, self).__init__(name, mac_ver)
+        name = mac_releases.get(str(version.up_to(part)), "macos")
+        super(MacOs, self).__init__(name, version.up_to(2))
```

Comment:
Actually this is **not** what we want. If we include the true minor version, then "monterey"-generated code on one machine may require a newer SDK than "monterey"-generated code on another machine. (Think what would happen if your binary cache–generating machine is 12.5 and you're running 12.4.) This kind of situation is what leads to thousands of warnings about mismatched SDK errors.